src/AutoRest.CSharp/MgmtTest/Generation/TestHelperWriter.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using AutoRest.CSharp.Generation.Writers;
+using AutoRest.CSharp.Mgmt.AutoRest;
+using AutoRest.CSharp.Output.Models.Types;
+
+namespace AutoRest.CSharp.MgmtTest.Generation
+{
+    internal class TestHelperWriter
+    {
+        private CodeWriter _writer;
+        private CodeWriter _tagsWriter = new CodeWriter();
+        private BuildContext<MgmtOutputLibrary> _context;
+
+        protected string TestNamespace => _context.DefaultNamespace + ".Tests";
+        protected string TypeNameOfThis => "TestHelper";
+
+        public TestHelperWriter(CodeWriter writer, BuildContext<MgmtOutputLibrary> context)
+        {
+            _writer = writer;
+            _context = context;
+        }
+
+        public void WriteMockExtension()
+        {
+            _writer.UseNamespace("System");
+            _writer.UseNamespace("System.Threading.Tasks");
+            _writer.UseNamespace($"{_context.DefaultNamespace}.Models");
+            _writer.UseNamespace("Azure.ResourceManager.Resources");
+            _writer.UseNamespace("Azure.ResourceManager.Resources.Models");
+
+            using (_writer.Namespace(TestNamespace))
+            {
+                _writer.WriteXmlDocumentationSummary($"Test Extension for {_context.DefaultNamespace}");
+                _writer.Append($"public static partial class {TypeNameOfThis:D}");
+                using (_writer.Scope())
+                {
+                    foreach (var resourceCollection in _context.Library.ResourceCollections)
+                    {
+                        var collectionWriter = new ResourceCollectionTestWriter(_writer, resourceCollection, _context);
+                        if (resourceCollection.CreateOperation is not null)
+                        {
+                            collectionWriter.WriteExampleInstanceMethod(resourceCollection.CreateOperation, true);
+                        }
+                        if (resourceCollection.GetOperation is not null)
+                        {
+                            collectionWriter.WriteExampleInstanceMethod(resourceCollection.GetOperation, true);
+                        }
+                        foreach (var clientOperation in resourceCollection.ClientOperations)
+                        {
+                            collectionWriter.WriteExampleInstanceMethod(clientOperation, true);
+                        }
+                    }
+
+                    WriteCreateResourceGroupMethod();
+                    WriteReplaceWith();
+                }
+            }
+        }
+
+        public void WriteCreateResourceGroupMethod()
+        {
+            _writer.UseNamespace("Azure.ResourceManager");
+            using (_writer.Scope($"public static async Task<ResourceGroup> CreateResourceGroupAsync(string resourceGroupName, ArmClient client)"))
```

Comment:
We can use the Type in the string interpolation to automatically use the corresponding namespaces, we do not need to use them explicitly
```suggestion
            using (_writer.Scope($"public static async Task<{typeof(ResourceGroup)}> CreateResourceGroupAsync(string resourceGroupName, {typeof(ArmClient)} client)"))
```